pkg/api/staking.go

```diff
@@ -0,0 +1,98 @@
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/gorilla/mux"
+)
+
+func (s *Service) stakingAccessHandler(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if !s.stakingSem.TryAcquire(1) {
+			s.logger.Debug("staking access: simultaneous on-chain operations not supported")
+			s.logger.Error(nil, "staking access: simultaneous on-chain operations not supported")
+			jsonhttp.TooManyRequests(w, "simultaneous on-chain operations not supported")
+			return
+		}
+		defer s.stakingSem.Release(1)
+
+		h.ServeHTTP(w, r)
+	})
+}
+
+type getStakeResponse struct {
+	StakedAmount *big.Int `json:"stakedAmount"`
+}
+
+func (s *Service) stakingDepositHandler(w http.ResponseWriter, r *http.Request) {
+	overlayAddr, err := hex.DecodeString(mux.Vars(r)["address"])
+	if err != nil {
+		s.logger.Debug("get stake: decode overlay address failed", "string", overlayAddr, "error", err)
+		s.logger.Error(nil, "get stake: decode overlay address failed")
+		jsonhttp.BadRequest(w, "invalid address")
+		return
+	}
+
+	stakedAmount, ok := big.NewInt(0).SetString(mux.Vars(r)["amount"], 10)
+	if !ok {
+		s.logger.Error(nil, "deposit stake: invalid amount")
+		jsonhttp.BadRequest(w, "invalid staking amount")
+		return
+	}
+	err = s.stakingContract.DepositStake(r.Context(), *stakedAmount, swarm.NewAddress(overlayAddr))
+	if err != nil {
+		if errors.Is(err, stakingcontract.ErrInvalidStakeAmount) {
+			s.logger.Debug("deposit stake: invalid stake amount", "error", err)
+			s.logger.Error(err, "deposit stake: invalid stake amount")
+			jsonhttp.BadRequest(w, "minimum 1 BZZ required for staking")
+			return
+		}
+		if errors.Is(err, stakingcontract.ErrNotImplemented) {
+			s.logger.Debug("deposit stake: not implemented", "error", err)
+			s.logger.Error(nil, "deposit stake: not implemented")
+			jsonhttp.NotImplemented(w, "not implemented")
+			return
+		}
+		if errors.Is(err, stakingcontract.ErrInsufficientFunds) {
+			s.logger.Debug("deposit stake: out of funds", "error", err)
+			s.logger.Error(nil, "deposit stake: out of funds")
+			jsonhttp.BadRequest(w, "out of funds")
+			return
+		}
+		s.logger.Debug("deposit stake: deposit failed", "error", err)
+		s.logger.Error(nil, "deposit stake: deposit failed")
+		jsonhttp.InternalServerError(w, "cannot stake")
+		return
+	}
+	jsonhttp.OK(w, nil)
+}
+
+func (s *Service) getStakedAmountHandler(w http.ResponseWriter, r *http.Request) {
+	overlayAddr, err := hex.DecodeString(mux.Vars(r)["address"])
+	if err != nil {
+		s.logger.Debug("get stake: decode overlay string failed", "string", overlayAddr, "error", err)
+		s.logger.Error(nil, "get stake: decode overlay string failed")
+		jsonhttp.BadRequest(w, "invalid address")
+		return
+	}
+
+	stakedAmount, err := s.stakingContract.GetStake(r.Context(), swarm.NewAddress(overlayAddr))
+	if err != nil {
+		s.logger.Debug("get stake: get staked amount failed", "overlayAddr", hex.EncodeToString(overlayAddr), "error", err)
```

Comment:
`swarm.Address` implements `Stringer`, so it can be logged using without `hex.EncodeToString()` - this only applies if address is created early on with `swarm.ParseHexAddress`